src/signature.c

```diff
@@ -24,198 +24,342 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
 #include <openssl/bio.h>
 #include <openssl/err.h>
 #include <openssl/pem.h>
+#include <openssl/crypto.h>
 
 #include "config.h"
 #include "signature.h"
 #include "swupd.h"
 
-/*
- * Implementation flavors:
- *   FAKE ..... do nothing, always return success
- *   FORGIVE .. do everything, always return success
- *   REAL ..... do everything, return the real status
- */
-#define IMPL_FAKE 0
-#define IMPL_FORGIVE 1
-#define IMPL_REAL 2
-
-#warning "TODO pick signing scheme"
-#if defined(SWUPD_LINUX_ROOTFS)
-#define IMPL IMPL_FAKE
+#define BUFFER_SIZE   4096
+#define CERTNAME "/usr/share/clear/update-ca/ClearLinuxRoot.pem"
```

Comment:
Again, I would prefer that this value be configurable at build time, but if you want to delay adding a new option (or consolidating with --enable-signature-verification), we can do that too.